app/models/zone.rb

```diff
@@ -7,6 +7,7 @@ class Zone < ApplicationRecord
   belongs_to :log_file_depot, :class_name => "FileDepot"
 
   has_many :miq_servers
+  has_many :active_miq_servers, -> { active_miq_servers }, :class_name => "MiqServer", :inverse_of => :zone, :dependent => nil
```

Comment:
the association is limited to the scope. so `active_miq_servers` is tacked into the `WHERE` clause

could have done `where(:active => true)`

Reply:
but this is a declaration _of_ active_miq_servers...won't it be an infinite loop?

Reply:
oh is the has_many declaring an instance method, whereas the scope is using a class method?

Reply:
confusing to the casual reader, that's for sure.  ;-)

Reply:
Since this is in the active_miq_server (`class_name: MiqServer`) relation, the scope is from `MiqServer` not on `Zone`.

We do not have good consistency with scope names.
I do see `with_`, `for_` and `is_` rare. Maybe if we changed the name of the scope it would read better:

```ruby
class MiqServer
  # was :active_miq_server
  scope :is_active, -> { where(:status => STATUSES_ACTIVE) }
end

class Zone
  has_many :active_miq_servers, -> { is_active }, :class_name => "MiqServer", :inverse_of => :zone, :dependent => nil
end
```